server/chat.ts

```diff
@@ -24,26 +24,59 @@ To reload chat commands:
 */
 
 'use strict';
-/** @typedef {GlobalRoom | GameRoom | ChatRoom} Room */
 
-/** @typedef {(this: PageContext, query: string[], user: User, connection: Connection) => (Promise<string | null | void> | string | null | void)} PageHandler */
-/** @typedef {{[k: string]: PageHandler | PageTable}} PageTable */
+type Room = import('./rooms').Room;
+type ChatRoom = import('./rooms').ChatRoomType;
+type GameRoom = import('./rooms').GameRoomType;
+type BasicChatRoom = import('./rooms').BasicChatRoomType;
 
-/** @typedef {(this: CommandContext, target: string, room: ChatRoom | GameRoom, user: User, connection: Connection, cmd: string, message: string) => (void)} ChatHandler */
-/** @typedef {{[k: string]: ChatHandler | string | true | string[] | ChatCommands}} ChatCommands */
+type User = import('./users').User;
```

Comment:
Ideally we have `Users` as a global type, as before.

Reply:
But I guess that's something I can just personally fix later.

Reply:
That just involves putting that typedef in `globals.ts`, right?

Reply:
Yeah.